Updated version

alepar666 opened the pull request:
- updated api for stations.json
- refactored parsing info
- removed 4th channel
- changed info background colour

alepar666 committed:
all thee stuuff

I hate git

superjohan commented on IFMNowPlaying.m:
```diff
@@ -23,11 +24,26 @@ @implementation IFMNowPlaying
+    NSLog(@"yearInt: %@",yearInt);
+    NSString *year = [NSString stringWithFormat:@"%@",yearInt];
+
+    NSString *nowPlaying1 = [dict[@"artist"] stringByAppendingString:[sepa1 stringByAppendingString:dict[@"track"]]];
```
I see what you meant in your email. :) This whole thing could probably be replaced with one `+[NSString stringWithFormat:]`, although that will change the logic a bit. As opposed to Java, in Objective-C passing messages to `nil` is okay and just does nothing. So when you append to `dict[@"artist"]`, it can be `nil` (however unlikely), and the resulting `nowPlaying5` will be `nil` too.

alepar666 replied:
I basically put together code I found around, it could be the worst piece of code I ever developed :| 
Anyway, the potential issue here is for sure if one of the JSON attributes is null or empty.
In my mind I would have developed something like
`String nowPlaying = "";
for(String field : fields){
 if(isNonEmpty(field)){
nowPlaying += field;
}
}

boolean isNonEmpty(String stringToBeChecked){
 return stringToBechecked != null && stringToBeChecked.length > 0;
}`

and then the separation management

superjohan replied:
Sure. If you can wait a few days, I'd be happy to work something out here. In the middle of xmas holidays with the family now. :)

alepar666 replied:
Of course! there is no rush, I am just working/playin on it since I am free now, but ready to revert all :)

superjohan commented:
Thanks for your PR! I'm waiting for my new Mac to arrive (should be here in a week or so), so I can't really take a very close look just yet, other than surface level code changes.

This PR is honestly a good reminder of the state I left the app in: `stations.json` was in place, but not building the actual UI based on the station config instead of using hardcoded images. You mentioned replacing the app with a PWA, which sounds like a better plan than working much more in this app.

superjohan committed:
Whitespace edit (as a test)

superjohan commented:
Yay I can apparently push commits to this branch. Fixes incoming

superjohan committed:
Add gitignore.

superjohan committed:
Remove unwanted files from the repo.

superjohan committed:
Fix infinite loop by stopping only if we're not already stopped.

(has this ever worked???)

superjohan committed:
Use tabs for indentation everywhere.

(not my preference these days but seeing as everything in this project is already indented with tabs, might as well make it formal)

superjohan committed:
Whitespace edits.

superjohan commented:
@alepar666: quick question: does the app actually work for you? :) I can't get streams to start, even after fixing the crashes...

alepar666 commented:
In this very moment streamings are down :\
https://www.intergalactic.fm/ <https://www.intergalactic.fm/>

Before yes was working perfectly

> Il giorno 13 gen 2021, alle ore 18:34, Johan Halin <notifications@github.com> ha scritto:
> 
> 
> @alepar666 <https://github.com/alepar666>: quick question: does the app actually work for you? :) I can't get streams to start, even after fixing the crashes...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/superjohan/ifm/pull/1#issuecomment-759604785>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ASH55PFSPP7IEBULBMG733DSZXKSJANCNFSM4VISYNPQ>.
>

superjohan commented:
Ahh okay. I got a little worried there. The app's insanely crash-happy too, fixing those right now.

alepar666 commented:
Well, a good test scenario. :)
By the way channel 3 is still up

> Il giorno 13 gen 2021, alle ore 18:41, Johan Halin <notifications@github.com> ha scritto:
> 
> 
> Ahh okay. I got a little worried there. The app's insanely crash-happy too, fixing those right now.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/superjohan/ifm/pull/1#issuecomment-759608431>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ASH55PEYX2GUCXOAGFPREQDSZXLLVANCNFSM4VISYNPQ>.
>

superjohan committed:
Less mental way of handling the various numbered views.

Must've thought I was so fuckin clever.

superjohan committed:
Add FIXME.

superjohan commented:
I did some edits, but didn't fix the Now Playing parsing yet. I want to do that on top of your other PR.
